File: infrastructure/branch_publisher_base/scripts/cache_exists.py
import os, sys, boto3
from botocore.exceptions import ClientError

# Check if yarn.lock file exists in cache for your JANIS_BRANCH

client = boto3.client('s3')
try:
    response = client.head_object(
        Bucket="coa-publisher-codebuild",
        Key=f'cache/{os.getenv("JANIS_BRANCH")}/yarn.lock'
    )
except ClientError as e:
    if e.__dict__['response']['Error']['Code'] == '404':
        print("False")
        sys.exit(0)
    else:
        raise e

exists = response['ResponseMetadata']['HTTPStatusCode'] == 200
print(exists)

File: src/janis_builder_base/scripts/cache_exists.js
const AWS = require('aws-sdk');

// Check if a yarn /cache directory exists in S3 for your Janis Branch.
// Returns true or false
async function cacheExists(janisBranch){
  const s3 = new AWS.S3();
  let headObject;

  try {
    const headObject = await s3.headObject({
      Bucket: "coa-publisher",
      Key: `cache/${janisBranch}/yarn.lock`,
    }).promise()

    if (headObject) {
      console.log("True")
    }
  } catch(e) {
    if (e.code === "NotFound") {
      console.log("False")
    } else {
      process.exit(1)
    }
  }
}

cacheExists(process.env.JANIS_BRANCH)
